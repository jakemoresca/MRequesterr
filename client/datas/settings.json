{
    "integrationSettings": {
        "movies" : {
            "baseUrl": "",
            "apiKey": "",
            "host": "",
            "port": 80,
            "useSsl": false
        },
        "series" : {
            "baseUrl": "",
            "apiKey": "",
            "host": "",
            "port": 80,
            "useSsl": false
        },
        "tmdbApiKey": ""
    }
}