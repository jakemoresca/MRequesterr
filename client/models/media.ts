import { IRadarrMovie, RadarrQueueRecord } from "./radarrMovies";
import { ISonarrSeries } from "./sonarrSeries";

export interface IMedia 
{
    id: string;
    tmdbId?: string;
    imdbId: string;
    tvdbId?: string;
    title: string;
    cleanTitle: string;
    sortTitle: string;
    status: string;
    overview: string;
    monitored: boolean;
    minimumAvailability: string;
    //profileId: number;
    runtime: number;
    images: IMediaImage[];
    titleSlug: string;
    path: string;
    year: number;
    genres: string[];
    isAvailable: boolean;
    hasFile: boolean;
    statistics: IStatistics;
    additionalInfo?: AdditionalMediaInfo;
    movieFile?: MovieFile;
    firstAired?: string;
}

export type AdditionalMediaInfo = ISonarrSeries | RadarrQueueRecord | IRadarrMovie | undefined;

export interface IMediaImage 
{
    coverType: string;
    url: string;
}

export interface IStatistics 
{
    percentOfEpisodes: number;
}

export interface MovieFile {
    movieId:             number;
    relativePath:        string;
    path:                string;
    size:                number;
    dateAdded:           Date;
    indexerFlags:        number;
    quality:             Welcome4Quality;
    mediaInfo:           MediaInfo;
    originalFilePath:    string;
    qualityCutoffNotMet: boolean;
    languages:           Language[];
    releaseGroup:        string;
    edition:             string;
    id:                  number;
}

export interface Language {
    id:   number;
    name: string;
}

export interface MediaInfo {
    audioBitrate:          number;
    audioChannels:         number;
    audioCodec:            string;
    audioLanguages:        string;
    audioStreamCount:      number;
    videoBitDepth:         number;
    videoBitrate:          number;
    videoCodec:            string;
    videoDynamicRangeType: string;
    videoFps:              number;
    resolution:            string;
    runTime:               string;
    scanType:              string;
    subtitles:             string;
}

export interface Welcome4Quality {
    quality:  QualityQuality;
    revision: Revision;
}

export interface QualityQuality {
    id:         number;
    name:       string;
    source:     string;
    resolution: number;
    modifier:   string;
}

export interface Revision {
    version:  number;
    real:     number;
    isRepack: boolean;
}
