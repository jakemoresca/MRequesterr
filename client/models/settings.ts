export interface ISettings
{
    integrationSettings: IIntegrationSettings;
}

export interface IIntegrationSettings
{
    movies: IMovieSettings;
    series: ISeriesSettings;
}

export interface IMovieSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}

export interface ISeriesSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}