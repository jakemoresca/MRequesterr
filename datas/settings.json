{
    "integrationSettings": {
        "movies": {
            "baseUrl": "/radarr",
            "apiKey": "",
            "host": "rydersir.polaris.usbx.me",
            "port": 443,
            "useSsl": true
        },
        "series": {
            "baseUrl": "/sonarr",
            "apiKey": "",
            "host": "rydersir.polaris.usbx.me",
            "port": 443,
            "useSsl": true
        },
        "auth": {
            "baseUrl": "/jellyfin",
            "apiKey": "",
            "host": "rydersir.polaris.usbx.me",
            "port": 443,
            "useSsl": true
        },
        "tmdbApiKey": ""
    },
    "plexLogin": true
}