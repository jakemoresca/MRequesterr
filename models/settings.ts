export interface ISettings
{
    integrationSettings: IIntegrationSettings;
    plexLogin: boolean;
}

export interface IIntegrationSettings
{
    movies: IMovieSettings;
    series: ISeriesSettings;
    auth: IAuthSettings;
}

export interface IMovieSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}

export interface ISeriesSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}

export interface IAuthSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}