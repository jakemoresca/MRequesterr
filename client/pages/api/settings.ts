// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
import type { NextApiRequest, NextApiResponse } from 'next'
import { ISettings } from '../../models/settings'

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ISettings>
) {
    const settings = await getSettings();
    res.status(200).json(settings)
}

export async function getSettings(): Promise<ISettings> {
    const settings = await import("../../datas/settings.json");

    settings.integrationSettings.movies.apiKey = process.env.RADARR_API_KEY ?? settings.integrationSettings.movies.apiKey;
    settings.integrationSettings.series.apiKey = process.env.SONARR_API_KEY ?? settings.integrationSettings.series.apiKey;
    settings.integrationSettings.auth.apiKey = process.env.JELLYFIN_API_KEY ?? settings.integrationSettings.auth.apiKey;

    return settings.default;
}
