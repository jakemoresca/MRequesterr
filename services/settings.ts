import { ISettings } from '../models/settings'

export async function getSettings(): Promise<ISettings> {
    const settings = await import("../datas/settings.json");

    settings.integrationSettings.movies.apiKey = process.env.RADARR_API_KEY ?? settings.integrationSettings.movies.apiKey;
    settings.integrationSettings.series.apiKey = process.env.SONARR_API_KEY ?? settings.integrationSettings.series.apiKey;
    settings.integrationSettings.auth.apiKey = process.env.JELLYFIN_API_KEY ?? settings.integrationSettings.auth.apiKey;
    settings.integrationSettings.discordWebhookUrl = process.env.DISCORD_WEBHOOK_URL ?? settings.integrationSettings.discordWebhookUrl;

    return settings.default;
}
