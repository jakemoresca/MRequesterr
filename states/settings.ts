import { atom } from "recoil";
import { ISettings } from "../models/settings";

export const settingsState = atom<ISettings>({
    key: 'settings', // unique ID (with respect to other atoms/selectors)
    default: undefined, // default value (aka initial value)
});