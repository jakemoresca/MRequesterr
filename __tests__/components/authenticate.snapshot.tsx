import { render } from '@testing-library/react'
import React from 'react';
import { RecoilRoot } from 'recoil';
import Authenticate from '../../components/authenticate';
import { ISettings } from '../../models/settings';

it('renders authenticate unchanged', () => {
  const settings: ISettings = {
    "integrationSettings": {
      "movies": {
        "baseUrl": "/radarr",
        "apiKey": "",
        "host": "test.web.com",
        "port": 443,
        "useSsl": true
      },
      "series": {
        "baseUrl": "/sonarr",
        "apiKey": "",
        "host": "test.web.com",
        "port": 443,
        "useSsl": true
      },
      "auth": {
        "baseUrl": "/jellyfin",
        "apiKey": "",
        "host": "test.web.com",
        "port": 443,
        "useSsl": true
      }
    }
  }

  const { container } = render(<RecoilRoot><Authenticate settings={settings} /></RecoilRoot>)
  expect(container).toMatchSnapshot()
})