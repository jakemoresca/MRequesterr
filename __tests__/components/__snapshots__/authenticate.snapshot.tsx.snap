// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders authenticate unchanged 1`] = `
<div>
  <div
    class="w-100 m-auto py-4 d-flex justify-content-center container"
  >
    <form
      action="/"
      class="col align-self-center"
      data-bitwarden-watching="1"
      style="max-width: 330px;"
    >
      <svg
        class="bi me-2"
        height="24"
        style="fill: white;"
        width="30"
      >
        <use
          xlink:href="#bootstrap"
        />
      </svg>
      <h1
        class="h3 mb-3 fw-normal"
      >
        Please sign in using your Jellyfin account
      </h1>
      <div
        class="form-floating"
      >
        <input
          class="form-control"
          id="floatingInput"
          placeholder="Username"
          type="text"
          value=""
        />
        <label
          for="floatingInput"
        >
          Username
        </label>
      </div>
      <div
        class="form-floating"
      >
        <input
          class="form-control"
          id="floatingPassword"
          placeholder="Password"
          type="password"
          value=""
        />
        <label
          for="floatingPassword"
        >
          Password
        </label>
      </div>
      <button
        class="w-100 btn btn-lg btn-primary"
        type="submit"
      >
        Sign in
      </button>
      <p
        class="mt-5 mb-3 text-muted"
      >
        Â© 2022
      </p>
    </form>
  </div>
</div>
`;
